### PR TITLE
Implement DPOS utility functions - Closes #6728

### LINKED ISSUE #6728: Implement util functions for DPoS module
### Description

Implement util functions from the old DPoS module

```
sortUnlocking
getMinPunishedHeight
getPunishmentPeriod
getMinWaitingHeight
getWaitingPeriod
isNullCharacterIncluded
isUsername
validateSignature
isCurrentlyPunished
```

### Acceptance Criteria

- All functions should be unit tested
- All logic exists in the previous DPoS module

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #6728

### How was it solved?

- Added functions: 
```sortUnlocking
getMinPunishedHeight
getPunishmentPeriod
getMinWaitingHeight
getWaitingPeriod
isNullCharacterIncluded
isUsername
validateSignature
isCurrentlyPunished
```
- Updated dpos v2 types
- Updated dpos v2 constants
### How was it tested?

Functions to be tested in hooks issues.

### CODE REVIEW

File: framework/src/modules/dpos_v2/types.ts [hunk 1]
```diff
@@ -55,3 +55,19 @@ export interface RandomAPI {
...
+	amount: bigint;
+}
+
+export interface UnlockingObject {
+	readonly delegateAddress: Buffer;
+	readonly amount: bigint;
+	readonly unvoteHeight: number;
+}
+
+export interface UnlockTransactionParamsContext {
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/types.ts:71]: I think this is not needed?

File: framework/src/modules/dpos_v2/types.ts [hunk 2]
```diff
@@ -55,3 +55,19 @@ export interface RandomAPI {
 export interface ValidatorsAPI {
 	setGeneratorList(apiContext: APIContext, generatorAddresses: Buffer[]): Promise<void>;
 }
+
+export interface VoteAccountObject {
```
Comment (shuse2) [inline framework/src/modules/dpos_v2/types.ts:59]: I think this is not needed

**Review by shuse2 — CHANGES_REQUESTED** [review]

**Review by shuse2 — APPROVED** [review]

File: framework/src/modules/dpos_v2/constants.ts [hunk 3]
```diff
@@ -26,3 +26,10 @@ export const COMMAND_ID_VOTE = 1;
 export const COMMAND_ID_UNLOCK = 2;
 export const COMMAND_ID_POM = 3;
 export const COMMAND_ID_UPDATE_GENERATOR_KEY = 4;
+
+export const WAIT_TIME_VOTE = 2000;
+export const WAIT_TIME_SELF_VOTE = 260000;
+export const VOTER_PUNISH_TIME = 260000;
+export const SELF_VOTE_PUNISH_TIME = 78000;
```
Comment (mehmetegemen) [inline framework/src/modules/dpos_v2/constants.ts:33]: This should be `780000` I think, one 0 is missing at the end.

**Review by mehmetegemen — CHANGES_REQUESTED** [review]

File: framework/src/modules/dpos_v2/constants.ts (hunk 3 above)
Comment (mitsuaki-u) [reply framework/src/modules/dpos_v2/constants.ts:33]: Good catch!

**Review by mehmetegemen — APPROVED** [review]

**Pull request merged**